### PR TITLE
feat(services): Add support for two factor authentication (SQSERVICES-1165)

### PR DESCRIPTION
**Pull request by thisisamir98**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQSERVICES-1165" title="SQSERVICES-1165" target="_blank">TaskSQSERVICES-1165</a>  [web] 2nd auth factor for "login"
  </td></table>
  <br />
 

…

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Implemented Two Factor Authentication

### CODE REVIEW

File: src/script/auth/page/Login.tsx [hunk 1]
```diff
@@ -214,47 +252,80 @@ const Login = ({
...
+                  <Label markInvalid={twoFactorSubmitFailed}>
+                    <CodeInput
+                      autoFocus
+                      style={{marginTop: 60}}
+                      onCodeComplete={submitTwoFactorLogin}
+                      data-uie-name="enter-code"
+                      markInvalid={twoFactorSubmitFailed}
+                    />
+                    {twoFactorSubmitFailed && (
+                      <div>{_(errorHandlerStrings[BackendError.LABEL.CODE_AUTHENTICATION_FAILED])}</div>
```
Comment (tlebon) [inline src/script/auth/page/Login.tsx]: i think i prefer to set this more dynamically in the error handling (based off the received error label) than to display it statically only in this case.

File: src/script/auth/page/Login.tsx [hunk 2]
```diff
@@ -182,11 +203,28 @@ const Login = ({
     }
   };
 
+  const resendTwoFactorCode = async () => {
+    setIsResendingTwoFactorCode(true);
```
Comment (tlebon) [inline src/script/auth/page/Login.tsx]: you should be able to infer this based off of the redux status
`startSendTwoFactorCode` && `successfulSendTwoFactorCode`

File: src/script/auth/page/Login.tsx [hunk 3]
```diff
@@ -163,6 +174,16 @@ const Login = ({
             history.push(ROUTE.CLIENTS);
             break;
           }
+          case BackendError.LABEL.CODE_AUTHENTICATION_REQUIRED: {
```
Comment (tlebon) [inline src/script/auth/page/Login.tsx:174]: keep in mind theres also a FE Feature flag for this, so possibly you would want to check if it is enabled. 

``# Feature toggle for 2nd factor authentication for some actions (ex. login)
FEATURE_ENABLE_2F_PASSWORD_CHALLENGE="false"``

File: src/script/auth/page/Login.tsx (hunk 2 above)
Comment (tlebon) [reply src/script/auth/page/Login.tsx]: from TM     `case UserActionCreator.successfulSendActivationCode.type: {
      return {...state, activatedEmail: action.payload, sendingCode: false};
    }`

Comment (thisisamir98) [reply src/script/auth/page/Login.tsx]: https://github.com/wireapp/wire-webapp/pull/12681/commits/feedbbcbeefcb509a19ce4e2b01e0ad51252cc26

File: src/script/auth/page/Login.tsx (hunk 1 above)
Comment (thisisamir98) [reply src/script/auth/page/Login.tsx]: https://github.com/wireapp/wire-webapp/pull/12681/commits/53093655850a363db3e6e280bbdd792d4afd993b

**Review by tlebon — APPROVED** [review]
LGTM!

**Pull request merged**